src/services/organizeImports.ts

```diff
@@ -254,9 +268,71 @@ namespace ts.OrganizeImports {
                 namedImports,
             };
         }
+    }
 
-        function compareIdentifiers(s1: Identifier, s2: Identifier) {
-            return compareStringsCaseInsensitive(s1.text, s2.text);
+    /* @internal */ // Internal for testing
+    /**
+     * @param exportGroup a list of ExportDeclarations, all with the same module name.
+     */
+    export function coalesceExports(exportGroup: ReadonlyArray<ExportDeclaration>) {
+        if (exportGroup.length === 0) {
+            return exportGroup;
+        }
+
+        const { exportWithoutClause, namedExports } = getCategorizedExports(exportGroup);
+
+        const coalescedExports: ExportDeclaration[] = [];
+
+        if (exportWithoutClause) {
+            coalescedExports.push(exportWithoutClause);
+        }
+
+        if (namedExports.length === 0) {
+            return coalescedExports;
+        }
+
+        const newExportSpecifiers: ExportSpecifier[] = [];
+        newExportSpecifiers.push(...flatMap(namedExports, i => (i.exportClause).elements));
+
+        const sortedExportSpecifiers = stableSort(newExportSpecifiers, (s1, s2) =>

```

Comment:
There's duplicate code in `coalesceImports`, could they share?

Reply:
I made `getCategorizedImports` shared and I find the result much uglier than what I started with.  Did you have something specific in mind?

Reply:
Just the `stableSort`:
```ts
function sortSpecifiers<T extends ImportSpecifier | ExportSpecifier>(specifiers: ReadonlyArray<T>): ReadonlyArray<T> {
    return stableSort(specifiers, (s1, s2) =>
        compareIdentifiers(s1.propertyName || s1.name, s2.propertyName || s2.name) ||
        compareIdentifiers(s1.name, s2.name));
}
```

Reply:
Done.